index.js

```diff
@@ -31,64 +31,62 @@ if(!heimdall.hasMonitor('babel')) {
   });
 }
 
-module.exports = Babel;
-function Babel(inputTree, _options) {
-  if (!(this instanceof Babel)) {
-    return new Babel(inputTree, _options);
-  }
-
-  let options = _options || {};
-  options.persist = 'persist' in options ? options.persist : true;
-  options.async = true;
-
-  if (options.browserPolyfill) {
-    let polyfillPath = require.resolve('@babel/polyfill/package.json');
-    polyfillPath = path.join(path.dirname(polyfillPath), "dist");
-
-    let browserPolyfillPath = options.browserPolyfillPath || "browser-polyfill.js"
-    let polyfill = funnel(polyfillPath, {
-      files: ['polyfill.js'],
-      getDestinationPath: () => browserPolyfillPath
-    });
-    inputTree = mergeTrees([polyfill, inputTree]);
-  }
-  delete options.browserPolyfill;
-  delete options.browserPolyfillPath;
+class Babel extends Filter {
+  constructor(inputTree, _options = {}) {
+    let options = _options;
+    options.persist = 'persist' in options ? options.persist : true;
+    options.async = true;
+
+    if (options.browserPolyfill) {
+      let polyfillPath = require.resolve('@babel/polyfill/package.json');
+      polyfillPath = path.join(path.dirname(polyfillPath), "dist");
+
+      let browserPolyfillPath = options.browserPolyfillPath || "browser-polyfill.js"
+      let polyfill = funnel(polyfillPath, {
+        files: ['polyfill.js'],
+        getDestinationPath: () => browserPolyfillPath
+      });
+      inputTree = mergeTrees([polyfill, inputTree]);
+    }
+    delete options.browserPolyfill;
+    delete options.browserPolyfillPath;
 
-  Filter.call(this, inputTree, options);
+    super(inputTree, options);
 
-  delete options.persist;
-  delete options.async;
-  delete options.annotation;
-  delete options.description;
+    delete options.persist;
+    delete options.async;
+    delete options.annotation;
+    delete options.description;
```

Comment:
For the 8.x, this should also be changed https://github.com/babel/broccoli-babel-transpiler/issues/189